remote.go

```diff
@@ -1032,7 +1032,7 @@ func (r *Remote) List(o *ListOptions) (rfs []*plumbing.Reference, err error) {
 
 	defer ioutil.CheckClose(s, &err)
 
-	ar, err := s.AdvertisedReferences()
+	ar, err := s.AdvertisedReferencesContext(context.TODO())
```

Comment:
Hi, I was using `go-git` package to do `git ls-remote` check. But I found this command doesn't have a timeout configuration. I saw you were doing this action. That would be nice! 

But why do you pass `context.TODO()` to it? As far as I know, `context.TODO()` ia an empty context. This means, `git ls-remote` also doesn't have timeout. Could you explain it for me pls? Thanks in advance!

Reply:
FetchContext and PushContext exist, but there's no ListContext method. That's a reasonable thing to add, it's just not part of this PR, which is doing the plumbing.

Reply:
Thanks for your reply! Could you please also help add `ListContext` func? Or may I submit a new PR based on your changes? But this requires your PR is merged, not sure how long it will take?

Reply:
I'd put the API change in a separate MR. No idea how long it will take somebody to get around to merging this one.

Reply:
Hope this PR can be merged soon!!!! By thw way, about API change PR, do you want to own it? If not, I can do it.

Reply:
Hi, could you please add a test as @mcuadros said so that this PR can be merged? Because in our environment, this feature is needed and lack of it iss causing some problems for us. Thanks in advance!

Reply:
ok, thanks!